### PR TITLE
Retry connect

### PR DESCRIPTION
**Pull request by Fizzadar**
1st commit does the dumb thing, 2nd takes it further but feels like we could/should abstract a retry method out in either a util or even mautrix-go? WDYT @tulir?

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -569,16 +585,20 @@ func (user *User) unlockedConnectWithCookies(cookies cookies.Cookies) (*messagix
 		return nil, fmt.Errorf("failed to get proxy: %w", err)
 	}
 	user.log.Debug().Msg("Connecting to Meta")
-	// TODO set proxy for media client?
-	cli, err := messagix.NewClient(MessagixPlatform, cookies, log, proxyAddr)
+	var cli *messagix.Client
+	err = doWithRetry(maxConnectAttempts, func() error {
+		// TODO set proxy for media client?
```
Comment (tulir) [inline user.go]: the TODO shouldn't be moved

File: user.go [hunk 2]
```diff
@@ -569,16 +585,20 @@ func (user *User) unlockedConnectWithCookies(cookies cookies.Cookies) (*messagix
...
 	if err != nil {
 		return nil, fmt.Errorf("failed to prepare client: %w", err)
 	}
 	if user.bridge.Config.Meta.GetProxyFrom != "" {
 		cli.GetNewProxy = user.getProxy
 	}
 	cli.SetEventHandler(user.eventHandler)
-	err = cli.Connect()
+	err = doWithRetry(maxConnectAttempts, cli.Connect)
 	if err != nil {
```
Comment (tulir) [inline user.go]: This one doesn't need a retry loop, it should never fail here

**Comment by Fizzadar** [comment]
Super quick @tulir 🙏! Updated

File: user.go [hunk 3]
```diff
@@ -570,10 +572,20 @@ func (user *User) unlockedConnectWithCookies(cookies cookies.Cookies) (*messagix
...
-	if err != nil {
-		return nil, fmt.Errorf("failed to prepare client: %w", err)
+	attempts := 1
+	var cli *messagix.Client
+	for {
+		cli, err = messagix.NewClient(MessagixPlatform, cookies, log, proxyAddr)
+		if err != nil {
+			if attempts > maxConnectAttempts {
+				return nil, fmt.Errorf("failed to prepare client: %w", err)
+			}
```
Comment (tulir) [inline user.go:595]: Oh right, this should also check that it's not a auth error (the ones listed in `Connect()`: token invalidated, challenge/consent required, LS error responses).

Or maybe check that it's a connection error specifically? That might be harder though

**Pull request merged**